pkg/tracing/jaeger/jaeger.go

```diff
@@ -33,6 +43,94 @@ func (t *Tracer) GetTraceIDFromSpanContext(ctx opentracing.SpanContext) (string,
 	return "", false
 }
 
+// NewTracerProvider returns a new instance of an OpenTelemetry tracer provider.
+func NewTracerProvider(ctx context.Context, logger log.Logger, conf []byte) (*tracesdk.TracerProvider, error) {
+	config := Config{}
+	if err := yaml.Unmarshal(conf, &config); err != nil {
+		return nil, err
+	}
+
+	var exporter *otel_jaeger.Exporter
+	var err error
+
+	if config.Endpoint != "" {
+		var jaegerCollectorEndpointOptions []otel_jaeger.CollectorEndpointOption
+		if config.User != "" {
+			jaegerCollectorEndpointOptions = append(jaegerCollectorEndpointOptions, otel_jaeger.WithUsername(config.User))
+		}
+		if config.Password != "" {
+			jaegerCollectorEndpointOptions = append(jaegerCollectorEndpointOptions, otel_jaeger.WithPassword(config.Password))
+		}
+		jaegerCollectorEndpointOptions = append(jaegerCollectorEndpointOptions, otel_jaeger.WithEndpoint(config.Endpoint))
+
+		exporter, err = otel_jaeger.New(otel_jaeger.WithCollectorEndpoint(jaegerCollectorEndpointOptions...))
+		if err != nil {
+			return nil, err
+		}
+	} else if config.AgentHost != "" && config.AgentPort != 0 {
+		var jaegerAgentEndpointOptions []otel_jaeger.AgentEndpointOption
+		jaegerAgentEndpointOptions = append(jaegerAgentEndpointOptions, otel_jaeger.WithAgentHost(config.AgentHost))
+		jaegerAgentEndpointOptions = append(jaegerAgentEndpointOptions, otel_jaeger.WithAgentPort(strconv.Itoa(config.AgentPort)))
+
+		exporter, err = otel_jaeger.New(otel_jaeger.WithAgentEndpoint(jaegerAgentEndpointOptions...))
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		exporter, err = otel_jaeger.New(nil)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
I added it since if neither endpoint nor agent is configured, it will rely on the default options for [collector endpoint](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/jaeger#WithCollectorEndpoint) and [agent](https://pkg.go.dev/go.opentelemetry.io/otel/exporters/jaeger#WithAgentEndpoint).

Reply:
I see now, maybe we could simplify this into a separate method which would looked at endpoint, agent host / port and returned one `EndpointOption` to use with `New` method?

Reply:
good idea, I'll do that for packages jaeger and OTLP. 
Thanks!